test/test.memo.js

```diff
@@ -0,0 +1,30 @@
+const tap = require('tap');
+const FunctionCache = require('../');
+
+const wait = (ms) => new Promise(resolve => setTimeout(resolve, ms));
+
+tap.test('memo', async t => {
+  let cache = new FunctionCache();
+  let called = 0;
+  const fP = async() => {
+    called++;
+    await wait(100);
+    return 21;
+  };
+  let r1 = await cache.memo('memo1', fP);
+  t.equal(called, 1);
+  t.equal(r1, 21);
+  r1 = await cache.memo('memo1', fP);
+  t.equal(called, 1);
+  t.equal(r1, 21);
+  cache = new FunctionCache(true);
+  r1 = await cache.memo('memo1', fP, 100, true);
+  t.equal(called, 2);
+  t.equal(r1, 21);
+  await wait(110);
+  r1 = await cache.memo('memo1', fP);
```

Comment:
can we make the function return a different value every time to make sure that it's actually working?